src/open_inwoner/accounts/tests/test_profile_views.py

```diff
@@ -1139,3 +1143,119 @@ def test_render_form_limit_newsletters_to_admin_selection(self, m):
+    def setUp(self):
+        super().setUp()
+
+        self.appointments_url = reverse("profile:appointments")
+        self.user = DigidUserFactory()
+
+        self.config = QmaticConfig.get_solo()
+        self.api_root = "https://qmatic.local/api/"
+        self.service = ServiceFactory.create(
+            api_root=self.api_root, api_type=APITypes.orc
+        )
+        self.config.service = self.service
+        self.config.save()
+
+        self.appointment_passport = AppointmentFactory.build(
+            title="Aanvraag paspoort",
+            start="2020-01-01T12:00:00+00:00",
+            notes="foo",
+            branch=BranchDetailFactory.build(
+                name="Hoofdkantoor",
+                timeZone="Europe/Amsterdam",
+                addressCity="Amsterdam",
+                addressLine2="Dam 1",
+            ),
+        )
+        self.appointment_idcard = AppointmentFactory.build(
+            title="Aanvraag ID kaart",
+            start="2020-03-06T16:30:00+00:00",
+            notes="bar",
+            branch=BranchDetailFactory.build(
+                name="Hoofdkantoor",
+                timeZone="America/New_York",
+                addressCity="New York",
+                addressLine2="Wall Street 1",
+            ),
+        )
+
+    def setUpMocks(self, m):
+        data = {
+            "notifications": [],
+            "meta": {
+                "start": "",
+                "end": "",
+                "totalResults": 1,
+                "offset": None,
+                "limit": None,
+                "fields": "",
+                "arguments": [],
+            },
+            "appointmentList": [
+                self.appointment_passport.dict(),
+                self.appointment_idcard.dict(),
+            ],
+        }
+        m.get(
+            f"{self.api_root}v1/customers/externalId/{self.user.email}/appointments",
+            json=data,
+        )
+
+    def test_do_not_render_list_if_config_is_missing(self, m):
+        self.config.service = None
+        self.config.save()
+
+        response = self.app.get(self.appointments_url, user=self.user)
+
+        self.assertIn(_("Geen afspraken beschikbaar"), response.text)
+
+    def test_do_not_render_list_if_no_appointments_are_found(self, m):
+        m.get(
+            f"{self.api_root}v1/customers/externalId/{self.user.email}/appointments",
+            status_code=404,
+        )
+
+        response = self.app.get(self.appointments_url, user=self.user)
+
+        self.assertIn(_("Geen afspraken beschikbaar"), response.text)
+
+    def test_render_list_if_appointments_are_found(self, m):
+        self.setUpMocks(m)
+
+        response = self.app.get(self.appointments_url, user=self.user)
+
+        self.assertIn(_("Een overzicht van uw afspraken"), response.text)
+
+        cards = response.pyquery(".appointment-info")
+
+        self.assertEqual(len(cards), 2)
+
+        passport_appointment = PQ(cards[0]).find("ul").children()
```

Comment:
Is it necessary to use PyQuery directly, since you're already using `response.pyquery`, or just more convenient?

Reply:
It actually is necessary, it's a bit weird but for some reason I can't use `.find` and other methods on the element that I get from `cards[0]` directly

Reply:
This looks familiar: iirc pyquery doesn't wrap recursively like some other api's do.